cmd/lifecycle/exporter.go

```diff
@@ -129,27 +129,49 @@ func (e *exportCmd) Args(nargs int, args []string) error {
 	}
 
 	var err error
-	e.stackMD, e.runImageRef, e.registry, err = resolveStack(e.imageNames[0], e.stackPath, e.runImageRef)
+	e.analyzedMD, err = parseAnalyzedMD(cmd.DefaultLogger, e.analyzedPath)
 	if err != nil {
-		return err
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
 	}
 
-	if e.runImageRef == "" {
-		return cmd.FailErrCode(
-			errors.New("-run-image is required when there is no stack metadata available"),
-			cmd.CodeInvalidArgs,
-			"parse arguments",
-		)
+	e.stackMD, err = readStack(e.stackPath)
+	if err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse stack metadata")
 	}
 
-	e.analyzedMD, err = parseOptionalAnalyzedMD(cmd.DefaultLogger, e.analyzedPath)
+	e.targetRegistry, err = parseRegistry(e.imageNames[0])
 	if err != nil {
-		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse target registry")
+	}
+
+	if err := e.populateRunImageRefIfNeeded(); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "populate run image")
 	}
 
 	return nil
 }
 
+func readStack(stackPath string) (platform.StackMetadata, error) {
+	var (
+		stackMD platform.StackMetadata
+		err     error
```

Comment:
Why do we need to define the error here?
I think you can just define it as part of the if condition on line 160.

Reply:
Updated in 07e41be